sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientBuilder.java

```diff
@@ -269,6 +289,18 @@ public HttpClient build() {
                     new ProxyAuthenticator(buildProxyOptions.getUsername(), buildProxyOptions.getPassword()));
             }
         }
+
+        // The JDK HttpClientBuilder only allows configuring a single choice.
+        // If HTTP/2 is chosen it will attempt to use it, if not available it will fall back to HTTP/1.1.
+        // If HTTP/1.1 is chosen it will not attempt to use HTTP/2.
+        // The default if version is not set is HTTP/2, which we don't want. So, if HTTP/2 is not in the set we
+        // set the version explicitly to HTTP/1.1.
+        if (protocolVersions != null && protocolVersions.contains(HttpProtocolVersion.HTTP_2)) {
+            httpClientBuilder.version(java.net.http.HttpClient.Version.HTTP_2);
+        } else {
+            httpClientBuilder.version(java.net.http.HttpClient.Version.HTTP_1_1);
+        }
```

Comment:
We should update the builder method documentation to call out that we use HTTP2 if both HTTP2 and HTTP1.1 are included in the EnumSet.